### DISCUSSION TITLE
NavLink triggers onClick when pressing on Chevron (@tanstack/router integration)

### DISCUSSION THREAD

**Discussion by jvllmr** [mantinedev/mantine#6463] [Q&A] [answered]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.10.2

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

When creating a Link the recommended way in @tanstack/router with mantine's NavLink (allowing @tanstack/router's powerful type inference) , @tanstack/router passes an `onClick` to NavLink. This results in a navigation when clicking on the chevron.

### If possible, include a link to a codesandbox with a minimal reproduction

https://stackblitz.com/edit/tanstack-router-8h1e6e?file=src%2Fmain.tsx&preset=node

### Possible fix

Don't pass the onClick to the entire NavLink, but only to the Label. Maybe also provide an `chevronOnClick` prop. Unfortunately, this would a breaking change for Users.

### Self-service

- [X] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
If you want to disable propagation of click events, use `event.stopPropagation` – https://codesandbox.io/p/sandbox/mantine-react-template-forked-ttp2ff?file=%2Fsrc%2FApp.tsx%3A17%2C43

**Comment by jvllmr** [comment, marked as answer]
Thanks for your Input. I refined your idea so that the rest of the component has the default anchor behaviour and the existence of children is checked.

```typescript
import { Box, NavLink, NavLinkProps } from '@mantine/core';
import { IconChevronRight } from '@tabler/icons-react';
import { createLink } from '@tanstack/react-router';
import React from 'react';
import { useState } from 'react';

export const MyNavLink = createLink(
  ({ children, ...props }: Omit<NavLinkProps, 'component' | 'opened'>) => {
    const [opened, setOpened] = useState(false);

    return (
      <NavLink
        opened={opened}
        {...props}
        rightSection={
          children ? (
            <Box
              component="button"
              onClick={(event) => {
                event.preventDefault();
                event.stopPropagation();

                setOpened((prev) => !prev);
              }}
            >
              <IconChevronRight
                size="0.8rem"
                stroke={1.5}
                className="mantine-rotate-rtl"
              />
            </Box>
          ) : undefined
        }
      >
        {children}
      </NavLink>
    );
  }
);
```
